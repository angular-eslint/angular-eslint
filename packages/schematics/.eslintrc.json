{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "no-restricted-imports": [
          "error",
          {
            "paths": [
              {
                "name": "@nx/devkit",
                "message": "Import from devkit-exports.ts instead of '@nx/devkit' directly."
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
