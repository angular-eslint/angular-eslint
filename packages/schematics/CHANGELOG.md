# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [12.3.1](https://github.com/angular-eslint/angular-eslint/compare/v12.3.0...v12.3.1) (2021-07-15)

**Note:** Version bump only for package @angular-eslint/schematics

# [12.3.0](https://github.com/angular-eslint/angular-eslint/compare/v12.2.2...v12.3.0) (2021-07-13)

### Features

- **schematics:** better support @angular/cli 12.1 ([#591](https://github.com/angular-eslint/angular-eslint/issues/591)) ([c5da07b](https://github.com/angular-eslint/angular-eslint/commit/c5da07b2d0c506dde24f0abc3e212db9deeaca82))

## [12.2.2](https://github.com/angular-eslint/angular-eslint/compare/v12.2.1...v12.2.2) (2021-07-10)

**Note:** Version bump only for package @angular-eslint/schematics

## [12.2.1](https://github.com/angular-eslint/angular-eslint/compare/v12.2.0...v12.2.1) (2021-07-10)

**Note:** Version bump only for package @angular-eslint/schematics

# [12.2.0](https://github.com/angular-eslint/angular-eslint/compare/v12.1.0...v12.2.0) (2021-06-20)

**Note:** Version bump only for package @angular-eslint/schematics

# [12.1.0](https://github.com/angular-eslint/angular-eslint/compare/v12.0.0...v12.1.0) (2021-05-30)

### Bug Fixes

- **schematics:** skip config for tsconfig.e2e.json when no e2e project is present ([#484](https://github.com/angular-eslint/angular-eslint/issues/484)) ([2673e59](https://github.com/angular-eslint/angular-eslint/commit/2673e59ec5e2708b7082fe7347ee1c96030ab6db))

### Features

- **schematics:** on `ng add` include a lint command if none exists ([#481](https://github.com/angular-eslint/angular-eslint/issues/481)) ([ae49af4](https://github.com/angular-eslint/angular-eslint/commit/ae49af4ae8af9fef306bda31e156ed4e5ddf058b))

# [12.0.0](https://github.com/angular-eslint/angular-eslint/compare/v4.3.0...v12.0.0) (2021-05-13)

### Features

- update eslint to ^7.26.0, [@typescript-eslint](https://github.com/typescript-eslint) to 4.23.0 ([9e31c38](https://github.com/angular-eslint/angular-eslint/commit/9e31c3881a13d6ce3b642b9c23c67e2e0f2d1aa1))
- update to angular v12 ([c80008d](https://github.com/angular-eslint/angular-eslint/commit/c80008df8f6b9d08daf3043dffc1be45f8cfbe81))
- update tslint-to-eslint-config to 2.4.0 ([7352ad2](https://github.com/angular-eslint/angular-eslint/commit/7352ad260644952abebf06773703f7b550d870fb))

# [4.3.0](https://github.com/angular-eslint/angular-eslint/compare/v4.2.1...v4.3.0) (2021-05-12)

**Note:** Version bump only for package @angular-eslint/schematics

## [4.2.1](https://github.com/angular-eslint/angular-eslint/compare/v4.2.0...v4.2.1) (2021-05-12)

**Note:** Version bump only for package @angular-eslint/schematics

# [4.2.0](https://github.com/angular-eslint/angular-eslint/compare/v4.1.0...v4.2.0) (2021-04-28)

### Features

- **schematics:** add add-eslint-to-project schematic ([#426](https://github.com/angular-eslint/angular-eslint/issues/426)) ([7ae557d](https://github.com/angular-eslint/angular-eslint/commit/7ae557d94f53833fbfbf5128d39f64c7bb1c3c5f))

# [4.1.0](https://github.com/angular-eslint/angular-eslint/compare/v4.0.0...v4.1.0) (2021-04-28)

### Bug Fixes

- **schematics:** support workspaces which use targets alias ([#424](https://github.com/angular-eslint/angular-eslint/issues/424)) ([da6bcf7](https://github.com/angular-eslint/angular-eslint/commit/da6bcf70027cc4b339ef2e825a931ec882d4711f))

### Features

- **schematics:** dynamically install tslint-to-eslint-config as needed ([#425](https://github.com/angular-eslint/angular-eslint/issues/425)) ([27ca474](https://github.com/angular-eslint/angular-eslint/commit/27ca474419defb79b552b233689a2dbf31b8ad92))

# [4.0.0](https://github.com/angular-eslint/angular-eslint/compare/v3.0.1...v4.0.0) (2021-04-18)

### Features

- **schematics:** options for convert-tslint-to-eslint ([#419](https://github.com/angular-eslint/angular-eslint/issues/419)) ([18fd863](https://github.com/angular-eslint/angular-eslint/commit/18fd863d6948578db96252da57702338a8ea5ea0))

## [3.0.1](https://github.com/angular-eslint/angular-eslint/compare/v3.0.0...v3.0.1) (2021-04-18)

**Note:** Version bump only for package @angular-eslint/schematics

# [3.0.0](https://github.com/angular-eslint/angular-eslint/compare/v2.1.1...v3.0.0) (2021-04-17)

### Features

- v3.0.0 ([#388](https://github.com/angular-eslint/angular-eslint/issues/388)) ([f92b184](https://github.com/angular-eslint/angular-eslint/commit/f92b184c5b0b57328d0a323ac8c89f1b3017b8d4))

## [2.1.1](https://github.com/angular-eslint/angular-eslint/compare/v2.1.0...v2.1.1) (2021-04-17)

**Note:** Version bump only for package @angular-eslint/schematics

# [2.1.0](https://github.com/angular-eslint/angular-eslint/compare/v2.0.2...v2.1.0) (2021-04-11)

**Note:** Version bump only for package @angular-eslint/schematics

## [2.0.2](https://github.com/angular-eslint/angular-eslint/compare/v2.0.1...v2.0.2) (2021-03-16)

**Note:** Version bump only for package @angular-eslint/schematics

## [2.0.1](https://github.com/angular-eslint/angular-eslint/compare/v2.0.0...v2.0.1) (2021-03-14)

**Note:** Version bump only for package @angular-eslint/schematics

# [2.0.0](https://github.com/angular-eslint/angular-eslint/compare/v1.2.0...v2.0.0) (2021-03-13)

### Features

- v2.0.0 ([#358](https://github.com/angular-eslint/angular-eslint/issues/358)) ([737fd04](https://github.com/angular-eslint/angular-eslint/commit/737fd04946a9533698c04665c771d944ffbe430c)), closes [#328](https://github.com/angular-eslint/angular-eslint/issues/328) [#245](https://github.com/angular-eslint/angular-eslint/issues/245)

### BREAKING CHANGES

- The format of results output has changed
- The rule no longer exists for use

- feat(template-parser): updated use of parseTemplate to improve loc data
- Requires @angular/compiler 11.2.0 and above

- feat(schematics): change way indent and quotes are handled by conversion schematics
- The conversion schematic handle these rules differently

# [1.2.0](https://github.com/angular-eslint/angular-eslint/compare/v1.1.0...v1.2.0) (2021-02-06)

**Note:** Version bump only for package @angular-eslint/schematics

# [1.1.0](https://github.com/angular-eslint/angular-eslint/compare/v1.0.0...v1.1.0) (2021-01-14)

**Note:** Version bump only for package @angular-eslint/schematics
