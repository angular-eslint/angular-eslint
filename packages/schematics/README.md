# @angular-eslint/schematics

Please see https://github.com/angular-eslint/angular-eslint for full usage instructions and guidance.

The `@angular-eslint/schematics` package is a set of custom Angular CLI Schematics which are used to add and update dependencies and configuration files which are relevant for running ESLint on an Angular workspace.
