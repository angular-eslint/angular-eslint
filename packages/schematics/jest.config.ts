/* eslint-disable */
'use strict';

export default {
  displayName: 'schematics',
  preset: '../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  testMatch: null,
  testRegex: ['./tests/.+\\.test\\.ts$'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/schematics',
  coverageReporters: ['text-summary', 'lcov'],
};
