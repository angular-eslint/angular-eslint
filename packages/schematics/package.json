{
  "name": "@angular-eslint/schematics",
  "version": "12.2.2",
  "description": "Angular Schematics for angular-eslint",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/angular-eslint/angular-eslint.git",
    "directory": "packages/schematics"
  },
  "files": [
    "dist",
    "package.json",
    "README.md",
    "LICENSE"
  ],
  "schematics": "./dist/collection.json",
  "ng-update": {
    "migrations": "./dist/migrations.json"
  },
  "ng-add": {
    "save": "devDependencies"
  },
  "dependencies": {
    "@angular-eslint/eslint-plugin": "12.2.2",
    "@angular-eslint/eslint-plugin-template": "12.2.2",
    "ignore": "5.1.8",
    "strip-json-comments": "3.1.1",
    "tmp": "0.2.1"
  },
  "devDependencies": {
    "@types/tmp": "0.2.0",
    "@typescript-eslint/experimental-utils": "4.28.2",
    "eslint": "^7.26.0",
    "tslint-to-eslint-config": "^2.4.0"
  },
  "peerDependencies": {
    "@angular/cli": ">= 12.0.0 < 13.0.0"
  },
  "gitHead": "d1d0e9f61f6a864d14507aa09c4b08c085309e7d"
}
