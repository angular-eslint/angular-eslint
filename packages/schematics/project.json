{
  "root": "packages/schematics",
  "type": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/workspace:run-commands",
      "outputs": ["packages/schematics/dist"],
      "options": {
        "cwd": "packages/schematics",
        "parallel": false,
        "commands": [
          "npx ncp ../../node_modules/@schematics/angular/application/schema.json src/application/schema.json",
          "npx ncp ../../node_modules/@schematics/angular/library/schema.json src/library/schema.json",
          "npx rimraf ./dist",
          "npx tsc -p tsconfig.json",
          "npx ncp src/collection.json dist/collection.json",
          "npx ncp src/migrations.json dist/migrations.json",
          "npx ncp src/ng-add/schema.json dist/ng-add/schema.json",
          "npx ncp src/convert-tslint-to-eslint/schema.json dist/convert-tslint-to-eslint/schema.json",
          "npx ncp src/application/schema.json dist/application/schema.json",
          "npx ncp src/library/schema.json dist/library/schema.json",
          "npx ncp src/workspace/schema.json dist/workspace/schema.json",
          "npx ncp src/add-eslint-to-project/schema.json dist/add-eslint-to-project/schema.json"
        ]
      }
    },
    "test": {
      "executor": "@nrwl/workspace:run-commands",
      "outputs": ["packages/schematics/coverage"],
      "options": {
        "cwd": "packages/schematics",
        "parallel": false,
        "commands": ["npx jest --coverage"]
      }
    },
    "clean": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "cwd": "packages/schematics",
        "parallel": false,
        "commands": ["npx rimraf ./dist"]
      }
    },
    "typecheck": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "cwd": "packages/schematics",
        "parallel": false,
        "commands": ["npx tsc -p tsconfig.json --noEmit"]
      }
    }
  },
  "implicitDependencies": []
}
