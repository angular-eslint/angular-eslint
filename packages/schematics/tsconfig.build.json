{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "baseUrl": ".",
    "paths": {
      "@angular-eslint/eslint-plugin": ["../../eslint-plugin/dist/index.js"],
      "@angular-eslint/eslint-plugin-template": [
        "../../eslint-plugin-template/dist/index.js"
      ]
    }
  },
  "include": ["src"]
}
