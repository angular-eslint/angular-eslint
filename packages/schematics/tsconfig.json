{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc/schematics"
  },
  "include": ["src/**/*"],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
