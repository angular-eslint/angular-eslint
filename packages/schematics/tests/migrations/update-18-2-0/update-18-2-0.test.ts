import { Tree } from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';

const migrationSchematicRunner = new SchematicTestRunner(
  '@angular-eslint/schematics',
  path.join(__dirname, '../../../src/migrations.json'),
);

describe('update-18-2-0', () => {
  let appTree: UnitTestTree;
  beforeEach(() => {
    appTree = new UnitTestTree(Tree.empty());
    appTree.create('package.json', JSON.stringify({}));
    appTree.create(
      'angular.json',
      JSON.stringify({
        $schema: './node_modules/@angular/cli/lib/config/schema.json',
        version: 1,
        newProjectRoot: 'projects',
        projects: {
          foo: {
            root: 'projects/foo',
          },
          bar: {
            root: 'projects/bar',
          },
        },
      }),
    );
  });

  it('should not update typescript-eslint and eslint dependencies if the project is not already on v8 prereleases of typescript-eslint', async () => {
    appTree.overwrite(
      'package.json',
      JSON.stringify({
        devDependencies: {
          '@typescript-eslint/eslint-plugin': '^7',
          '@typescript-eslint/utils': '^7',
          '@typescript-eslint/parser': '^7',
          eslint: '^8.28.0',
        },
      }),
    );

    const tree = await migrationSchematicRunner.runSchematic(
      'update-18-2-0',
      {},
      appTree,
    );
    const packageJSON = JSON.parse(tree.readContent('/package.json'));
    expect(packageJSON).toMatchSnapshot();
  });

  it('should update typescript-eslint and eslint dependencies if the project is already on v8 prereleases of typescript-eslint', async () => {
    appTree.overwrite(
      'package.json',
      JSON.stringify({
        devDependencies: {
          '@typescript-eslint/eslint-plugin': '8.0.0-alpha.50',
          'typescript-eslint': '8.0.0-alpha.50',
          '@typescript-eslint/utils': '8.0.0-alpha.50',
          '@typescript-eslint/parser': '8.0.0-alpha.50',
          '@typescript-eslint/type-utils': '8.0.0-alpha.50',
          '@typescript-eslint/rule-tester': '8.0.0-alpha.50',
          eslint: '^9.1.0',
        },
      }),
    );

    const tree = await migrationSchematicRunner.runSchematic(
      'update-18-2-0',
      {},
      appTree,
    );
    const packageJSON = JSON.parse(tree.readContent('/package.json'));
    expect(packageJSON).toMatchSnapshot();
  });
});
