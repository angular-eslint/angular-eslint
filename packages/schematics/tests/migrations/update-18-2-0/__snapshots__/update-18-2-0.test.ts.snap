// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-18-2-0 should not update typescript-eslint and eslint dependencies if the project is not already on v8 prereleases of typescript-eslint 1`] = `
Object {
  "devDependencies": Object {
    "@typescript-eslint/eslint-plugin": "^7",
    "@typescript-eslint/parser": "^7",
    "@typescript-eslint/utils": "^7",
    "eslint": "^8.28.0",
  },
}
`;

exports[`update-18-2-0 should update typescript-eslint and eslint dependencies if the project is already on v8 prereleases of typescript-eslint 1`] = `
Object {
  "devDependencies": Object {
    "@typescript-eslint/eslint-plugin": "^8.0.0",
    "@typescript-eslint/parser": "^8.0.0",
    "@typescript-eslint/rule-tester": "^8.0.0",
    "@typescript-eslint/type-utils": "^8.0.0",
    "@typescript-eslint/utils": "^8.0.0",
    "eslint": "^9.8.0",
    "typescript-eslint": "^8.0.0",
  },
}
`;
